android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt

```diff
@@ -113,17 +162,88 @@ abstract class BaseModule(context: ReactApplicationContext?) : ReactContextBaseJ
         }
     }
 
+    override fun onFinished(result: SessionPaymentResult) {
+        sendFinishEvent(
+            SessionDetailsResponse(
+                result.sessionData ?: "",
+                "",
+                result.resultCode,
+                null,
+                result.sessionResult,
+                result.order
+            )
+        )
+    }
+
     override fun onAdditionalData(jsonObject: JSONObject) {
         sendEvent(DID_PROVIDE, jsonObject)
     }
 
+    override fun canHandleAction(action: Action): Boolean {
+        throw NotImplementedError("This is base class")
+    }
+
+    override fun handleAction(action: Action, activity: Activity) {
+        throw NotImplementedError("This is base class")
+    }
+
+    override fun handleIntent(intent: Intent) {
+        val redirectHandler = DefaultRedirectHandler()
+        val json = redirectHandler.parseRedirectResult(intent.data)
+        val httpClient = HttpClientFactory.getHttpClient(environment)
+        val sessionService = SessionService(httpClient)
+
+        CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+            .launch {
+                session?.let {
+                    val request = SessionDetailsRequest(
+                        sessionData = it.sessionSetupResponse.sessionData,
+                        paymentData = null,
+                        details = json
+                    )
+                    val response = sessionService.submitDetails(
```

Comment:
Do we need to handle errors here like no connectivity?

Reply:
Fixed on [bb620ea](https://github.com/Adyen/adyen-react-native/pull/331/commits/bb620eae941fd6dbc55953346324c6d58e06357e)